# 看文献，做笔记

[A Galaxy-based training resource for single-cell RNA-sequencing quality control and analyses](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6905351/)
+ Gigascience/2019 Dec; 8(12): giz144.
+ scater v1.10.1 
+ Galaxy
+ PCA: principal component analysis/t-SNE: t-stochastic neighbour embedding.
+ QC 

[scruff: an R/Bioconductor package for preprocessing single-cell RNA-sequencing data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6498700/)
+ BMC Bioinformatics/2019 May 2
+ Scruff:an R/Bioconductor package that preprocesses data generated from the CEL-Seq or CEL-Seq2 protocols and reports comprehensive data quality metrics and visualizations. scruff rapidly demultiplexes, aligns, and counts the reads mapped to genome features with deduplication of unique molecular identifier (UMI) tags. scruff also provides novel and extensive functions to visualize both pre- and post-alignment data quality metrics for cells from multiple experiments. The package also supports the visualization of quality metrics for sequence alignment files for multiple experiments generated by Cell Ranger from 10X Genomics. scruff is available as a free and open-source R/Bioconductor package.
+ preprocessing
+ It performs data demultiplexing, alignment, counting, quality report and visualization systematically and comprehensively, ensuring reproducible and reliable analysis of scRNA-seq data.

[pyNVR: investigating factors affecting feature selection from scRNA-seq data for lineage reconstruction](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6596893/)
+ Bioinformatics/2019 Jul 1
+ pyNVR
+ NCBI GEO repository
+ This package works as an unsupervised feature selection pipeline on quality controlled scRNA-seq data. scRNA-seq count or FPKM data act as inputs into this pipeline. 

[Quality control of single-cell RNA-seq by SinQC](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4978927/)
+ Bioinformatics/2016 Aug 15
+ SinQC：a method and software tool to detect technical artifacts in scRNA-seq samples by integrating both gene expression patterns and data quality information. We apply SinQC to nine different scRNA-seq datasets, and show that SinQC is a useful tool for controlling scRNA-seq data quality.



[A statistical simulator scDesign for rational scRNA-seq experimental design](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6612870/)
+ Bioinformatics/2019 Jul 5
+ scDesign:the first statistical framework for researchers to quantitatively assess practical scRNA-seq experimental design in the context of differential gene expression analysis. 
+ scDesign is expected to be an effective bioinformatic tool that assists rational scRNA-seq experimental design and comparison of scRNA–seq computational methods based on specific research goals.


[Random forest based similarity learning for single cell RNA sequencing data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6022547/)
+ Bioinformatics/2018 Jul 1
+ RAFSIL:a two-step approach for learning similarities between single cells based on whole transcriptome sequencing data.
+ RAFSIL implements a two-step procedure, first feature construction, and then similarity learning using random forests (RFs); it is flexible and easy to modify, expand and optimize.
+ We show RAFSIL learns similarities that faithfully represent group structure in scRNA-seq data; when used for dimension reduction and clustering they provide an accurate visualization of datasets and enable exploratory analyses for cell type identification and discovery. Importantly, RAFSIL compares favorably with the current state-of-the-art showing high accuracy and robustness, and we demonstrate how it enables the identification of technical variation that remains hidden with other approaches.

[Performance Assessment and Selection of Normalization Procedures for Single-Cell RNA-Seq](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6544759/)
+ Cell Systems/ 24 April 2019
SCONE(Bioconductor R software package)
![](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6544759/bin/nihms-1525125-f0001.jpg)
+ The goal of normalization is to account for observed differences in measurements between samples and/or features (e.g., genes) resulting from technical artifacts or unwanted biological effects (e.g., batch effects) rather than biological effects of interest. To derive gene expression measures from single-cell RNA sequencing (scRNA-seq) data and subsequently compare these measures between cells, analysts must normalize read counts (or other expression measures) to adjust for obvious differences in sequencing depths. When there are other significant biases in expression quantification, it may be necessary to further adjust expression measures for more complex unwanted technical factors related to sample and library preparation.
+ The full “scone” pipeline consists of two steps prior to normalization, namely (1) QC assessment and (2) optional sample and gene filtering, resulting in the selection of high-quality cell gene expression profiles.
+ scone uses a two-part normalization template to define an ensemble of normalization procedures: (1) scaling of the counts to account for between-sample differences in sequencing depth and other parameters of the read count distributions; (2) regression-based adjustment for known unwanted factors, such as processing batches, and unknown unwanted factors

[ClusterMap: compare multiple single cell RNA-Seq datasets across different experimental conditions](https://pubmed.ncbi.nlm.nih.gov/30649203/)
+ Bioinformatics/2019 Sep 1
+ ClusterMap is implemented in R
+ compare multiple single cell RNA-Seq datasets across different experimental conditions

[DoubletDecon: Deconvoluting Doublets from Single-Cell RNA-Sequencing Data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6983270/)
+ Cell Reports/5 November 2019
+ DoubletDecon:an approach that detects doublets with a combination of deconvolution analyses and the identification of unique cell-state gene expression.

[Granatum: a graphical single-cell RNA-Seq analysis pipeline for genomics scientists](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5716224/)
+ Genome Medicine/05 December 2017
+ Granatum, a web-based scRNA-Seq analysis pipeline to make analysis more broadly accessible to researchers.
+ It has a comprehensive list of modules, including plate merging and batch-effect removal, outlier-sample removal, gene-expression normalization, imputation, gene filtering, cell clustering, differential gene expression analysis, pathway/ontology enrichment analysis, protein network interaction visualization, and pseudo-time cell series construction.


[scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6105007/)
+ PLoS Comput Biol.2018 Aug 10
+ scPipe, an R/Bioconductor package that integrates barcode demultiplexing, read alignment, UMI-aware gene-level quantification and quality control of raw sequencing data generated by multiple protocols that include CEL-seq, MARS-seq, Chromium 10X, Drop-seq and Smart-seq.
+ scPipe produces a count matrix that is essential for downstream analysis along with an HTML report that summarises data quality. These results can be used as input for downstream analyses including normalization, visualization and statistical testing. 


[DECENT: differential expression with capture efficiency adjustmeNT for single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/pmid/31197307/)
+ Bioinformatics.2019 Jun 14
+ DECENT, a method for DE analysis of scRNA-seq data that explicitly and accurately models the molecule capture process in scRNA-seq experiments.
+ We build a capture model to account for the gene- and cell-specific properties of molecule capturing. This allows us to perform DE analysis on the inferred pre-dropout distributions of RNA molecules.We named our method Differential Expression with Capture Efficiency adjustmeNT (DECENT). 

[Spectral clustering based on learning similarity matrix](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454479/)
+ Bioinformatics. 2018 Feb 8
+ Single-cell RNA-sequencing (scRNA-seq) technology can generate genome-wide expression data at the single-cell levels. One important objective in scRNA-seq analysis is to cluster cells where each cluster consists of cells belonging to the same cell type based on gene expression patterns.
+ MATLAB
+ a novel spectral clustering algorithm



[DIMM-SC: a Dirichlet mixture model for clustering droplet-based single cell transcriptomic data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454475/)
+ Bioinformatics /2018 Jan 1
DIMM-SC:a Dirichlet Mixture Model for clustering droplet-based Single Cell transcriptomic data
+ We developed DIMM-SC, a Dirichlet Mixture Model for clustering droplet-based Single Cell transcriptomic data. 
+ We performed comprehensive simulations to evaluate DIMM-SC and compared it with existing clustering methods such as K-means, CellTree and Seurat. In addition, we analyzed public scRNA-Seq datasets with known cluster labels and in-house scRNA-Seq datasets from a study of systemic sclerosis with prior biological knowledge to benchmark and validate DIMM-SC. Both simulation studies and real data applications demonstrated that overall, DIMM-SC achieves substantially improved clustering accuracy and much lower clustering variability compared to other existing clustering methods. More importantly, as a model-based approach, DIMM-SC is able to quantify the clustering uncertainty for each single cell, facilitating rigorous statistical inference and biological interpretations, which are typically unavailable from existing clustering methods.


[Using neural networks for reducing the dimensions of single-cell RNA-Seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5737331/)
+ Nucleic Acids Res. 2017 Sep 29
+ These include questions about the best methods for clustering scRNA-Seq data, how to identify unique group of cells in such experiments, and how to determine the state or function of specific cells based on their expression profile. 
+ neural networks (NN) 

[JingleBells: A Repository of Immune-Related Single-Cell RNA–Sequencing Datasets](https://www.jimmunol.org/content/198/9/3375.long)
+ J Immunol May 1, 2017
+ To facilitate datasets reuse, we created the JingleBells repository for immune-related scRNA-seq datasets ready for analysis and visualization of reads at the single-cell level 
+ In this study, we report the formatting of publicly available scRNA-seq datasets into the standard binary sequence alignment/map (BAM) format (10), extended by the read-to-cell association, and grouped by experimental condition (performing the above preprocessing steps 1–5). The formatted datasets are deposited in the + + JingleBells repository. The repository and the uniform extended BAM format enable users without any programming skills to download datasets and visualize the reads aligned to specific genes of interest (or genomic regions) in Integrative Genomics Viewer (IGV), grouped and colored by their cell of origin. This visualization enables appreciation of the uniformity of coverage across the gene and the splicing patterns and genetic variation displayed by each cell. Moreover, computational biologists can easily reuse JingleBells’ formatted scRNA-seq datasets for further analysis, without the laborious preprocessing of each dataset.

[Scedar: A scalable Python package for single-cell RNA-seq exploratory data analysis](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7217489/)
+ PLoS Comput Biol. 2020 Apr
+ scedar, a scalable Python package for scRNA-seq exploratory data analysis. The package provides a convenient and reliable interface for performing visualization, imputation of gene dropouts, detection of rare transcriptomic profiles, and clustering on large-scale scRNA-seq datasets.

[CIDR: Ultrafast and accurate clustering through imputation for single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5371246/)
+ Genome Biol. 2017
+ we introduce CIDR (Clustering through Imputation and Dimensionality Reduction), an ultrafast algorithm that uses a novel yet very simple implicit imputation approach to alleviate the impact of dropouts in scRNA-seq data in a principled manner. Using a range of simulated and real data, we show that CIDR improves the standard principal component analysis and outperforms the state-of-the-art methods, namely t-SNE, ZIFA, and RaceID, in terms of clustering accuracy. 

[Falco: a quick and flexible single-cell RNA-seq processing framework on the cloud](https://academic.oup.com/bioinformatics/article/33/5/767/2559427)
+ Bioinformatics/1 March 2017
+ Falco, a cloud-based framework to enable paralellization of existing RNA-seq processing pipelines using big data technologies of Apache Hadoop and Apache Spark for performing massively parallel analysis of large scale transcriptomic data.

[scTyper: a comprehensive pipeline for the cell typing analysis of single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7430822/)
+ BMC Bioinformatics/2020 Aug 4
+ Recent advances in single-cell RNA sequencing (scRNA-seq) technology have enabled the identification of individual cell types, such as epithelial cells, immune cells, and fibroblasts, in tissue samples containing complex cell populations.
+ scTyper:a comprehensive and user-friendly R-based scRNA-seq analysis and cell typing package
+ scTyper also provides a database of cell type markers, scTyper.db, which contains 213 cell marker sets collected from literature. 
+ The package also supports the data preprocessing pipelines by Cell Ranger from 10X Genomics and the Seurat package. 

[bayNorm: Bayesian gene expression recovery, imputation and normalization for single-cell RNA-sequencing data](https://academic.oup.com/bioinformatics/article/36/4/1174/5581401)
+ Bioinformatics/15 February 2020
+ bayNorm, a novel Bayesian approach for scaling and inference of scRNA-seq counts.
+ bayNorm improves accuracy and sensitivity of differential expression analysis and reduces batch effect compared with other existing methods. 
+ bayNorm provides an efficient, integrated solution for global scaling normalization, imputation and true count recovery of gene expression measurements from scRNA-seq data.

[Accurate estimation of cell composition in bulk expression through robust integration of single-cell information](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7181686/)
+ Nat Commun. 2020 Apr 24
+ Bisque, a tool for estimating cell type proportions in bulk expression. Bisque implements a regression-based approach that utilizes single-cell RNA-seq (scRNA-seq) or single-nucleus RNA-seq (snRNA-seq) data to generate a reference expression profile and learn gene-specific bulk expression transformations to robustly decompose RNA-seq data.

[Systematic determination of the mitochondrial proportion in human and mice tissues for single-cell RNA sequencing data quality control](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btaa751/5896986?redirectedFrom=fulltext)
+ Bioinformatics/ 25 August 2020
+ Quality control (QC) is a critical step in single-cell RNA-seq (scRNA-seq) data analysis.
+ An important QC metric is the mitochondrial proportion (mtDNA%), which is used as a threshold to filter out low-quality cells.
+ Early publications in the field established a threshold of 5% and since then, it has been used as a default in several software packages for scRNA-seq data analysis, and adopted as a standard in many scRNA-seq studies.
+ In general, for mouse tissues, the 5% threshold performs well to distinguish between healthy and low-quality cells. However, for human tissues, the 5% threshold should be reconsidered as it fails to accurately discriminate between healthy and low-quality cells in 29.5% (13 of 44) tissues analyzed. We conclude that omitting the mtDNA% QC filter or adopting a suboptimal mtDNA% threshold may lead to erroneous biological interpretations of scRNA-seq data.

[scHinter: imputing dropout events for single-cell RNA-seq data with limited sample size](https://academic.oup.com/bioinformatics/article/36/3/789/5544928)
+ Bioinformatics/1 February 2020
+ However, scRNA-seq data suffer from problems of extremely high dropout rate and cell-to-cell variability, demanding new methods to recover gene expression loss.
+ We proposed a method called scHinter for imputing dropout events for scRNA-seq with special emphasis on data with limited sample size.

[DENDRO: genetic heterogeneity profiling and subclone detection by single-cell RNA sequencing](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6961311/)
+ Genome Biol. 
+ DENDRO, an analysis method for scRNA-seq data that clusters single cells into genetically distinct subclones and reconstructs the phylogenetic tree relating the subclones. 
+ DENDRO utilizes transcribed point mutations and accounts for technical noise and expression stochasticity. 

[alona: a web server for single-cell RNA-seq analysis](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7320629/)
+ Bioinformatics. 2020 Jun 15
+ The widespread adoption of scRNA-seq has created a need for user-friendly software for data analysis. 
+ alona can perform quality filtering, normalization, batch correction, clustering, cell type annotation and differential gene expression analysis. Data are visualized in the web browser using an interface based on JavaScript, allowing the user to query genes of interest and visualize the cluster structure. alona accepts a compressed gene expression matrix and identifies cell clusters with a graph-based clustering strategy. Cell types are identified from a comprehensive collection of marker genes or by specifying a custom set of marker genes.


[QUBIC2: a novel and robust biclustering algorithm for analyses and interpretation of large-scale RNA-Seq data ](https://academic.oup.com/bioinformatics/article/36/4/1143/5567116)
+ Bioinformatics/15 February 2020


[Putative cell type discovery from single-cell gene expression data](https://www.nature.com/articles/s41592-020-0825-9)
+ Nature Methods volume /18 May 2020
+ SCCAF:is based on the iterative application of machine learning and self-projection14 to clusters, thus gradually merging clusters that can be defined by similar sets of feature genes.

[Using multiple measurements of tissue to estimate subject- and cell-type-specific gene expression](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7523682/)
+ Bioinformatics/2020 Feb 1;
+ R package MIND
+ we develop a method to estimate subject- and cell-type-specific (CTS) gene expression from tissue using an empirical Bayes method that borrows information across multiple measurements of the same tissue per subject (e.g. multiple regions of the brain). 


[scds: computational annotation of doublets in single-cell RNA sequencing data ](https://academic.oup.com/bioinformatics/article/36/4/1150/5566507)
+ Bioinformatics/ 10 September 2019
+ However, it is known that these methods sometimes wrongly consider two or more cells as single cells, and that a number of so-called doublets is present in the output of such experiments.
+ we propose two new approaches for in silico doublet identification: Co-expression based doublet scoring (cxds) and binary classification based doublet scoring (bcds). 

[Quantifying pluripotency landscape of cell differentiation from scRNA-seq data by continuous birth-death process](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6876891/)
+ PLoS Comput Biol. 2019 Nov 13
+ We show that LDD can accurately and efficiently build the evolution tree of cells with pseudo-time, in particular quantifying their differentiation landscape in terms of potency. Taken together, this study provides not only a computational tool to quantify cell potency based on scRNA-seq data, but also a theoretical approach to understand the cell differentiation process from a dynamic perspective.

[SSCC: A Novel Computational Framework for Rapid and Accurate Clustering Large-scale Single Cell RNA-seq Data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6624216/)
+ Genomics Proteomics Bioinformatics/2019 Jun 13
+ Spearman subsampling-clustering-classification (SSCC), a new clustering framework based on random projection and feature construction, for large-scale scRNA-seq data. 
+ SSCC greatly improves clustering accuracy, robustness, and computational efficacy for various state-of-the-art algorithms benchmarked on multiple real datasets. 

[Evaluating measures of association for single-cell transcriptomics](https://www.nature.com/articles/s41592-019-0372-4)
+ Nature Methods /08 April 2019
+ However, owing to the unique statistical properties of scRNA-seq data, the optimal measures of association for identifying gene–gene and cell–cell relationships from single-cell transcriptomics remain unclear. 


[PROSSTT: probabilistic simulation of single-cell RNA-seq data for complex differentiation processes](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6748774/)
+ Bioinformatics. 2019 Sep 15
+ PROSSTT can simulate scRNA-seq datasets for differentiation processes with lineage trees of any desired complexity, noise level, noise model and size. PROSSTT also provides scripts to quantify the quality of predicted lineage trees.


[scBFA: modeling detection patterns to mitigate technical noise in large-scale single-cell genomics data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6734238/)
+ Genome Biol.  2019 Sep 9
+ scBFA, for both cell type identification and trajectory inference. Performance gains can also be realized in one line of R code in existing pipelines.

[LAmbDA: label ambiguous domain adaptation dataset integration reduces batch effects and improves subtype detection](https://academic.oup.com/bioinformatics/article-abstract/35/22/4696/5481958?redirectedFrom=fulltext)
+ Bioinformatics/ 30 April 2019

[ASAP: a web-based platform for the analysis and interactive visualization of single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5870842/)
+ Bioinformatics. 2017 Oct 1
+ We developed a fully integrated, web-based platform aimed at the complete analysis of scRNA-seq data post genome alignment: from the parsing, filtering and normalization of the input count data files, to the visual representation of the data, identification of cell clusters, differentially expressed genes (including cluster-specific marker genes), and functional gene set enrichment. This Automated Single-cell Analysis Pipeline (ASAP) combines a wide range of commonly used algorithms with sophisticated visualization tools.

[Accounting for technical noise in differential expression analysis of single-cell RNA sequencing data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5737676/)
+ Nucleic Acids Res. 2017 Nov 2
+ To account for cell-to-cell technical differences, we propose a statistical framework, TASC (Toolkit for Analysis of Single Cell RNA-seq), an empirical Bayes approach to reliably model the cell-specific dropout rates and amplification bias by use of external RNA spike-ins. TASC incorporates the technical parameters, which reflect cell-to-cell batch effects, into a hierarchical mixture model to estimate the biological variance of a gene and detect differentially expressed genes. More importantly, TASC is able to adjust for covariates to further eliminate confounding that may originate from cell size and cell cycle differences. In simulation and real scRNA-seq data, TASC achieves accurate Type I error control and displays competitive sensitivity and improved robustness to batch effects in differential expression analysis, compared to existing methods. 



[OEFinder: a user interface to identify and visualize ordering effects in single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4848403/)
+ Bioinformatics. 2016 May 1
+ OEFinder is available as an R package along with user-friendly graphical interface implementations which allows users to check for potential artifacts in scRNA-seq data generated by the Fluidigm C1 platform.

[SINCERA: A Pipeline for Single-Cell RNA-Seq Profiling Analysis](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4658017/)
+ PLoS Comput Biol. 2015 Nov 24
+ The pipeline supports the analysis for: 1) the distinction and identification of major cell types; 2) the identification of cell type specific gene signatures; and 3) the determination of driving forces of given cell types. 

[Integrating single-cell transcriptomic data across different conditions, technologies, and species](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6700744/)
+ Seurat：业界大佬，名气很大

[Exploring the single-cell RNA-seq analysis landscape with the scRNA-tools database](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6034903/)
+ PLoS Comput Biol/ 2018 Jun 25
+ Navigating the vast sea of tools now available is becoming increasingly challenging for researchers. In order to better facilitate selection of appropriate analysis tools we have created the scRNA-tools database (www.scRNA-tools.org) to catalogue and curate analysis tools as they become available. Our database collects a range of information on each scRNA-seq analysis tool and categorises them according to the analysis tasks they perform.

[ascend: R package for analysis of single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6735844/)
+ Gigascience.2019 Aug 24
+ The package incorporates both novel and established methods to provide a framework to perform cell and gene filtering, quality control, normalization, dimension reduction, clustering, differential expression, and a wide range of visualization functions.

[Evaluation of tools for highly variable gene discovery from single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6781572/)


[Bias, robustness and scalability in single-cell differential expression analysis](https://www.nature.com/articles/nmeth.4612)



[MetaCell: analysis of single-cell RNA-seq data using K-nn graph partitions](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6790056/)


[Latent cellular analysis robustly reveals subtle diversity in large-scale single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6902034/)
+ Nucleic Acids Res/2019 Sep 30
+ Latent Cellular Analysis (LCA), a machine learning–based analytical pipeline that combines cosine-similarity measurement by latent cellular states with a graph-based clustering algorithm. LCA provides heuristic solutions for population number inference, dimension reduction, feature selection, and control of technical variations without explicit gene filtering. We show that LCA is robust, accurate, and powerful by comparison with multiple state-of-the-art computational methods when applied to large-scale real and simulated scRNA-seq data.

[Reconstructing complex lineage trees from scRNA-seq data using MERLoT](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6755101/)
+ Nucleic Acids Res. 2019 Sep 26
+ MERLoT(an R package) (https://github.com/soedinglab/merlot), a flexible and user-friendly tool to reconstruct complex lineage trees from single-cell transcriptomics data. 
+ MERLoT allows users to easily retrieve subpopulations of cells that belong to specific branches or belong to specific paths along the tree. It can also calculate pseudotime assignments, impute pseudotemporal gene expression profiles or find genes that are differentially expressed on different tree segments.

[A Bayesian mixture model for clustering droplet-based single-cell transcriptomic data from population studies](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6456731/)
+ Nat Commun. 2019
+ a Bayesian mixture model for single-cell sequencing (BAMM-SC) method to cluster scRNA-seq data from multiple individuals simultaneously. BAMM-SC takes raw count data as input and accounts for data heterogeneity and batch effect among multiple individuals in a unified Bayesian hierarchical model framework. 

[Cerebro: interactive visualization of scRNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7141853/)
+ Bioinformatics. 2020 Apr 1
+ few tools exist that allow biologists without extensive bioinformatic expertise to directly visualize and interact with their own data and results. 
Cerebro (cell report browser), a Shiny- and Electron-based standalone desktop application for macOS and Windows which allows investigation and inspection of pre-processed single-cell transcriptomics data without requiring bioinformatic experience of the user. 
+ Through an interactive and intuitive graphical interface, users can (i) explore similarities and heterogeneity between samples and cell clusters in two-dimensional or three-dimensional projections such as t-SNE or UMAP, (ii) display the expression level of single genes or gene sets of interest, (iii) browse tables of most expressed genes and marker genes for each sample and cluster and (iv) display trajectories calculated with Monocle 2.

[Scater: pre-processing, quality control, normalization and visualization of single-cell RNA-seq data in R](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5408845/)
+ Bioinformatics. 2017 Apr 15
+ preparing raw sequence data for further analysis is not a straightforward process. Biases, artifacts and other sources of unwanted variation are present in the data, requiring substantial time and effort to be spent on pre-processing, quality control (QC) and normalization.
+ We have developed the R/Bioconductor package scater to facilitate rigorous pre-processing, quality control, normalization and visualization of scRNA-seq data. 

[An Efficient and Flexible Method for Deconvoluting Bulk RNA-Seq Data with Single-Cell RNA-Seq Data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6830085/)
+ Cells. 2019 Oct
+ Here, we developed a computationally statistical method, referring to Multi-Omics Matrix Factorization (MOMF), to estimate the cell-type compositions of bulk RNA sequencing (RNA-seq) data by leveraging cell type-specific gene expression levels from single-cell RNA sequencing (scRNA-seq) data. MOMF not only directly models the count nature of gene expression data, but also effectively accounts for the uncertainty of cell type-specific mean gene expression levels.
+ In this paper, we present two improvements for the application of t-SNE to scRNA-seq data visualization. First, we present FFT-accelerated Interpolation-based t-SNE (FIt-SNE), an algorithm for rapid computation of one- and two-dimensional t-SNE based on polynomial interpolation and further accelerated using the fast Fourier transform. We also present t-SNE heatmaps, a heatmap-style visualization method based on one-dimensional t-SNE, which simultaneously visualizes expression patterns of hundreds to thousands of genes.


[Cell composition analysis of bulk genomics using single cell data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6443043/)
+ Nat Methods. 2019 Apr
+ Here we introduce Cell Population Mapping (CPM), a deconvolution algorithm in which the composition of cell types and states is inferred from the bulk transcriptome using reference scRNA-seq profiles ('scBio' CRAN R-package). 


[Fast Interpolation-based t-SNE for Improved Visualization of Single-Cell RNA-Seq Data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6402590/)
+ Nat Methods. 2019 Mar
+ We dramatically accelerate t-SNE, obviating the need for data downsampling, and hence allowing visualization of rare cell populations.


[DeepImpute: an accurate, fast, and scalable deep neural network method to impute single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6798445/)
Genome Biol.2019 Oct 18
DeepImpute, a deep neural network-based imputation algorithm that uses dropout layers and loss functions to learn patterns in the data, allowing for accurate imputation. 


[DEsingle for detecting three types of differential expression in single-cell RNA-seq data](https://academic.oup.com/bioinformatics/article/34/18/3223/4983067)
Bioinformatics/24 April 2018
 R package DEsingle
The excessive amount of zeros in single-cell RNA-seq (scRNA-seq) data includes ‘real’ zeros due to the on-off nature of gene transcription in single cells and ‘dropout’ zeros due to technical reasons. Existing differential expression (DE) analysis methods cannot distinguish these two types of zeros. We developed an R package DEsingle which employed Zero-Inflated Negative Binomial model to estimate the proportion of real and dropout zeros and to define and detect three types of DE genes in scRNA-seq data with higher accuracy.


[A web server for comparative analysis of single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6233170/)
Nat Commun.  2018 Nov 13
We developed an automated pipeline to download, process, and annotate publicly available scRNA-seq datasets to enable large scale supervised characterization. We extend supervised neural networks to obtain efficient and accurate representations for scRNA-seq data. We apply our pipeline to analyze data from over 500 different studies with over 300 unique cell types and show that supervised methods outperform unsupervised methods for cell type identification. A case study highlights the usefulness of these methods for comparing cell type distributions in healthy and diseased mice. Finally, we present scQuery, a web server which uses our neural networks and fast matching methods to determine cell types, key genes, and more.


